### PR TITLE
[SPARK-12231][SQL]create a combineFilters' projection when we call buildPartitionedTableScan

### PR DESCRIPTION
**Pull request by kevinyu98**
Hello Michael & All: Here I am submitting another approach to solve this problem. Can you verify ? 

I think the problem is related to change from spark-10829,

before that PR change, the projects and filters are done inside buildPartitionedTableScan.
With that PR change, the filter expression divide to 3 parts, the filter left outside of the scan (combineFilters ) needs a different projection. 

So the fix is to create a combine projection for the outside filter and beyond. 

Thanks for your comments.

### CODE REVIEW

File: sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/DataSourceStrategy.scala [hunk 1]
```diff
@@ -85,9 +85,11 @@ private[sql] object DataSourceStrategy extends Strategy with Logging {
         s"Selected $selected partitions out of $total, pruned $percentPruned% partitions."
       }
 
+      // need to add projections from combineFilters in
+      val combineProjections = projects.toSet.union(combineFilters.flatMap(_.references).toSet).toSeq
```
Comment (marmbrus) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/DataSourceStrategy.scala]: We use use an `AttributeSet` anytime we are dealing with `AttributeReference` objects.

**Comment by marmbrus** [comment]
ok to test

**Comment by kevinyu98** [comment]
Hello Michael: I fixed the scala style issue, can you help re-run the test? Thanks.

**Comment by marmbrus** [comment]
ok to test

**Comment by kevinyu98** [comment]
The failure is because of the changed project, will submit an updated patch tomorrow.

**Comment by kevinyu98** [comment]
I will create a new PR.

**Pull request closed without merging**